services/core-api/app/api/compliance/resources/compliance_article_update_resource.py

```diff
@@ -0,0 +1,40 @@
+from flask_restx import Resource, inputs, reqparse, fields
+from marshmallow.exceptions import MarshmallowError
+from werkzeug.exceptions import BadRequest
+
+from app.api.compliance.models.compliance_article import ComplianceArticle
+from app.api.compliance.response_models import COMPLIANCE_ARTICLE_MODEL, COMPLIANCE_ARTICLE_UPDATE_MODEL
+from app.api.utils.resources_mixins import UserMixin
+from app.extensions import api
+from app.api.utils.access_decorators import EDIT_CODE, requires_any_of
+
+
+class ComplianceArticleUpdateResource(Resource, UserMixin):
+    parser = reqparse.RequestParser()
+
+    parser.add_argument(
+        'compliance_article_codes',
+        type=list,
+        location='json',
+        store_missing=False,
+        required=True
+    )
+
+    @api.doc(description='Update an existing Compliance Article.', body=COMPLIANCE_ARTICLE_UPDATE_MODEL)
+    @api.expect(parser)
+    @api.marshal_with(COMPLIANCE_ARTICLE_MODEL, code=200, as_list=True)
+    @requires_any_of([EDIT_CODE])
+    def put(self):
+        data = self.parser.parse_args()
+        compliance_article_codes = data.get('compliance_article_codes')
+        compliance_article = ComplianceArticle()
+        try:
+            update_compliance_article_codes = compliance_article.update_all(compliance_article_codes)
+
+            if len(update_compliance_article_codes) == len(compliance_article_codes):
+                compliance_article.save_all(update_compliance_article_codes)
+
+            return update_compliance_article_codes
+        except MarshmallowError as e:
+            raise BadRequest(f'Something went wrong while trying to update {e}')
```

Comment:
Not sure about returning {e} in the error message if we're trying to hide SQL/system errors. But a console error would be handy for debugging if something goes wrong in prod.

Reply:
Updated. I removed it and added it to logger error.